### DISCUSSION TITLE
Audio over OSC

### DISCUSSION THREAD

**Discussion by cipp0** [pschatzmann/arduino-audio-tools#1468] [Ideas]
Since my knowledge of CPP is very limited I wanted to propose the possibility of using this protocol as future developments: [https://git.iem.at/cm/aoo/-/tree/esp32](https://git.iem.at/cm/aoo/-/tree/esp32)

there is already a branch regarding the ESP32, but I don't know how to integrate it :)

**Comment by pschatzmann** [comment]
I considered adding AOO but for some reasons, I never found the time or found it too complicated. I don't remember the reasons why I did not advance on this.

Maybe something for some cold winter nights...

**Comment by Spacechild1** [comment]
Hi, AOO author. Just saw this by accident.

@cipp0 AOO is already a C/C++ library that can be used on all kinds of systems, from desktop to embedded. I'm not really sure what you expect @pschatzmann to do.

> there is already a branch regarding the ESP32, but I don't know how to integrate it :)

Here's a full example for using AOO on an ESP32: https://git.iem.at/ressi/esp32_aoo_test/. (I want to add ESP32 and Android examples to the "official" AOO repository for the next [pre]release.)

**Reply by cipp0** [reply]
The funny thing is that I am a student at IEM and i talked about this with Winfried Ritsch some months ago! ahahahah so, i would say, let's meet in Graz :)

**Reply by Spacechild1** [reply]
Haha small world :) To be fair, the ESP32 example didn't exist yet at the time you've opened this issue.
